martin/src/mbtiles/mod.rs

```diff
@@ -31,7 +55,31 @@ impl ConfigExtras for MbtConfig {
 
 impl SourceConfigExtras for MbtConfig {
     async fn new_sources(&self, id: String, path: PathBuf) -> FileResult<TileInfoSource> {
-        Ok(Box::new(MbtSource::new(id, path).await?))
+        let source = MbtSource::new(id, path.clone()).await?;
+        if let Err(validation_error) = source.validate(self.validate).await {
+            match self.on_invalid {
+                OnInvalid::Abort => {
+                    return Err(FileError::AbortOnInvalid(
+                        path,
+                        validation_error.to_string(),
+                    ));
+                }
+                OnInvalid::Ignore => {
+                    return Err(FileError::IgnoreOnInvalid(
+                        path,
+                        validation_error.to_string(),
+                    ));
+                }
+                OnInvalid::Warn => {
```

Comment:
We might want to do further processing of the `validation_error` to determine if this is a critical issue or not.

Reply:
I keep wondering if "fast" and "skip" are actually the same thing. We should always do some basic validation - like making sure the `tiles` table exists with the right columns - or else we can't even use it. The only difference is in what we do afterwards - ignore the source or continue with it...  So perhaps:

* validation level is `fast` vs `thorough` (no skip)
* severity would be `good`, `warn`, `error`
* and the action could be `abort` or `ignore source` for `error`, and `abort` or `ignore source` or `warn` on `warning`?
* we may even introduce `warnings-as-errors` parameter? (slowly building a compiler here :) )